test/fgpu/test_engine.py

```diff
@@ -112,30 +118,31 @@ def frac_channel(output: Output, channel: float) -> float:
 class TestEngine:
     r"""Grouping of unit tests for :class:`.Engine`\'s various functionality."""
 
-    @pytest.fixture(
-        params=[
-            pytest.param(
-                f"--wideband=name=test_stream,dst=239.10.11.0+15:7149,taps={TAPS}",
-                id="wideband",
-            ),
-            # Centre frequency is a multiple of the channel width
-            pytest.param(
-                f"--narrowband=name=test_stream,dst=239.10.11.0+15:7149,taps={TAPS},"
-                "decimation=8,centre_frequency=339077148.4375",
-                id="narrowband",
-            ),
-        ]
-    )
-    def output_arg(self, channels: int, request) -> str:
-        return f"{request.param},channels={channels}"
+    @pytest.fixture
+    def wideband_args(self, channels: int) -> str:
+        """Arguments to pass to the command-line parser for the wideband output."""
+        return f"{WIDEBAND_ARGS},channels={channels}"
 
     @pytest.fixture
-    def output(self, output_arg: str) -> Output:
-        arg_name, arg_value = output_arg.split("=", 1)
-        if arg_name == "--wideband":
-            return parse_wideband(arg_value)
+    def narrowband_args(self, channels: int) -> str:
+        """Arguments to pass to the command-line parser for the narrowband output."""
+        return f"{NARROWBAND_ARGS},channels={channels}"
+
+    @pytest.fixture(params=["wideband", "narrowband"])
+    def output(self, wideband_args: str, narrowband_args: str, request: pytest.FixtureRequest) -> Output:
+        """The output to run tests against."""
+        if request.param == "wideband":
+            return parse_wideband(wideband_args)
         else:
-            return parse_narrowband(arg_value)
+            return parse_narrowband(narrowband_args)
```

Comment:
So if I'm understanding correctly, this test is more or less the same as it previously was, except that
- it's now repeated twice,
- now both wide and narrowband are running, but
- only one output is checked each time.

Do I have it more or less correct?

Reply:
Yes.